### PR TITLE
refactor(role): determine level of role entity

### PR DESCRIPTION
**Pull request by apoorvdixit88**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Add entity type to roles (predefined and customs)
This can be used to determine at which level a current entity exist, and can perform operations accordingly.

### Additional Changes

- [ ] This PR modifies the API contract
- [x] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
Closes [#5487](https://github.com/juspay/hyperswitch/issues/5487)

## How did you test it?
The change is at DB level and can be tested with the upcoming application changes PRs.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: migrations/2024-07-30-124102_add_entity_type_to_roles/up.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+-- Your SQL goes here
+ALTER TABLE roles ADD COLUMN entity_type VARCHAR(64);
```
Comment (ThisIsMani) [inline migrations/2024-07-30-124102_add_entity_type_to_roles/up.sql:2]: It need not to be nullable, we can put default as merchant.

**Comment by ThisIsMani** [comment]
Schema consistency is failing for `v2`. Run `just migrate_v2 run` and commit the new schema file.

File: migrations/2024-07-30-124102_add_entity_type_to_roles/up.sql (hunk 1 above)
Comment (ThisIsMani) [reply migrations/2024-07-30-124102_add_entity_type_to_roles/up.sql:2]: No, we don't have to remove the default at all.

When we insert the roles we are anyways always inserting the entity_type of the role.

Putting it default will remove the burden of running the backfill queries. Like we did for version.

**Review by ThisIsMani — APPROVED** [review]
I think it's better to add comments wherever there will be changes when we remove V1.

**Review by racnan — APPROVED** [review]

**Review by jarnura — COMMENTED** [review]
We can make the DB layer also enum instead of varchar.

**Review by jarnura — APPROVED** [review]

**Comment by apoorvdixit88** [comment]
> We can make the DB layer also enum instead of varchar.

In `user_roles` table also we have kept it varchar . We can cover both in separate PR.

**Pull request merged**